cypress/e2e/products/images.js

```diff
@@ -44,54 +42,121 @@ xdescribe("Tests for images", () => {
-          .get(PRODUCT_DETAILS.productImage)
-          .find("img")
-          .invoke("attr", "src");
-      })
-      .then(imageUrl => {
-        cy.request(imageUrl);
-      })
-      .then(imageResp => {
-        expect(imageResp.status).to.equal(200);
-      });
-  });
+      deleteProductsStartsWith(name);
+      deleteCollectionsStartsWith(name);
+      cy.clearSessionData().loginUserViaRequest();
+      createNewProductWithNewDataAndDefaultChannel({ name })
+        .then(({ product }) => {
+          cy.visit(productDetailsUrl(product.id))
+            .waitForProgressBarToNotBeVisible()
+            .get(PRODUCT_DETAILS.uploadImageButton)
+            .click()
+            .get(PRODUCT_DETAILS.uploadSavedImagesButton)
+            .click()
+            .get(SHARED_ELEMENTS.fileInput)
+            .attachFile("images/saleorDemoProductSneakers.png")
+            .get(PRODUCT_DETAILS.productImage)
+            .find("img")
+            .invoke("attr", "src");
+        })
+        .then(imageUrl => {
+          cy.request(imageUrl);
+        })
+        .then(imageResp => {
+          expect(imageResp.status).to.equal(200);
+        });
+    },
+  );
+
+  it(
+    "should create thumbnail url after entering image url",
+    { tags: ["@products", "@allEnv", "@stable"] },
+    () => {
+      const failedRequests = [];
+      getFirstProducts(100)
+        .then(products => {
+          const productsWithThumbnails = products.filter(
+            product => product.node.thumbnail !== null,
+          );
+          productsWithThumbnails.forEach(product => {
+            const thumbnailUrl = product.node.thumbnail.url;
+            if (!thumbnailUrl.includes("/media/thumbnails")) {
+              cy.request({
+                url: thumbnailUrl,
+                followRedirect: false,
+                failOnStatusCode: false,
+              }).then(response => {
+                if (response.status !== 302) {
+                  failedRequests.push(
+                    `product: ${product.node.id}, expectedStatus: 302, status: ${response.status}`,
+                  );
+                } else {
+                  cy.request({
+                    url: response.redirectedToUrl,
+                    failOnStatusCode: false,
+                  }).then(resp => {
+                    if (resp.status !== 200) {
+                      failedRequests.push(
+                        `product: ${product.node.id}, expectedStatus: 200, status: ${resp.status}`,
+                      );
+                    }
+                  });
+                }
+              });
+            } else {
+              cy.request({ url: thumbnailUrl, failOnStatusCode: false }).then(
+                response => {
+                  if (response.status !== 200) {
+                    failedRequests.push(
+                      `product: ${product.node.id}, expectedStatus: 200, status: ${response.status}`,
+                    );
+                  }
```

Comment:
Response handling logic is the same, maybe let's make a function before 112 line eg. `handleResponse(expectedStatus, receivedStatus)`?